internal/pkg/client/cache/dir.go

```diff
@@ -22,42 +23,186 @@ const (
+	// It was in the past a global variable that was set when calling cache
+	// functions and updated based on the value of the environment variable.
+	// This approach created problem in multi-threaded cases (e.g., unit
+	// tests) where each thread was concurrently updating the environment
+	// variable creating inconsistencies.
+	rootDir string
+
+	// Library provides the location of the Library cache
+	Library string
+
+	// OciTemp provides the location of the OciTemp cache
+	OciTemp string
+
+	// OciBlob provides the location of the OciBlob cache
+	OciBlob string
+
+	// Net provides the location of the Net cache
+	Net string
+
+	// Shub provides the location of the Shub cache
+	Shub string
+
+	// Oras provides the location of the ORAS cache
+	Oras string
+}
+
+// HdlInit initializes a new cache within a given directory. It does not set
+// the environment variable to specify the location of the cache, the caller
+// being in charge of doing so. This also allows us to have a thread-safe
+// function, changing the value of the environment variable potentially
+// impacting other threads (e.g., while running unit tests). If baseDir is an
+// empty string, the image cache will be located to the default location, i.e.,
+// $HOME/.singularity.
+func HdlInit(baseDir string) (*ImgCache, error) {
+	if baseDir == "" {
+		baseDir = getCacheBasedir()
+	}
+
+	rootDir := getCacheRoot(baseDir)
+
+	// We make sure that the rootDir is actually a valid value
+	// FIXME: not really necessary anymore
+	user, err := user.Current()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user information: %s", err)
+	}
+	if rootDir == "" || rootDir == user.HomeDir {
+		return nil, fmt.Errorf("invalid root directory")
+	}
+
+	if err = initCacheDir(rootDir); err != nil {
+		return nil, fmt.Errorf("failed initializing caching directory: %s", err)
+	}
+
+	newCache := new(ImgCache)
+	newCache.BaseDir = baseDir
+	newCache.rootDir = rootDir
+	newCache.Library, err = getLibraryCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Library cache: %s", err)
+	}
+	newCache.OciTemp, err = getOciTempCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the OCI temp cache")
+	}
+	newCache.OciBlob, err = getOciBlobCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the OCI blob cache")
+	}
+	newCache.Net, err = getNetCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Net cache")
+	}
+	newCache.Shub, err = getShubCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Shub cache")
+	}
+	newCache.Oras, err = getOrasCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the ORAS cache")
+	}
+
+	return newCache, nil
+}
+
+// getCacheBaseDir figures out where the cache base directory is.
+// Singularity makes the following assumptions:
```

Comment:
the formatter for the Go documentation will consider everything before a blank line a single paragraph, so this would get formatted in a way similary to this:

> getCacheBaseDir figures out where the cache base directory is. Singularity makes the following assumptions: - a user ...

just add a blank between the first and second lines (but include the leading //)